src/test/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtilTest.java

```diff
@@ -0,0 +1,109 @@
+
+        List<Dataverse> dataverses = new ArrayList<>();
+        String publishedDvString = "publishedDv1";
+        Dataverse publishedDataverse = new Dataverse();
+        publishedDataverse.setAlias(publishedDvString);
+        SimpleDateFormat dateFmt = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date dvModifiedDate = dateFmt.parse("1955-11-12 22:04:00");
+        publishedDataverse.setModificationTime(new Timestamp(dvModifiedDate.getTime()));
+        publishedDataverse.setPublicationDate(new Timestamp(dvModifiedDate.getTime()));
+        dataverses.add(publishedDataverse);
+
+        List<Dataset> datasets = new ArrayList<>();
+
+        Dataset published = new Dataset();
+        String publishedPid = "doi:10.666/FAKE/published1";
+        published.setGlobalId(new GlobalId(publishedPid));
+        published.setPublicationDate(new Timestamp(new Date().getTime()));
+        published.setModificationTime(new Timestamp(new Date().getTime()));
+        datasets.add(published);
+
+        Dataset unpublished = new Dataset();
+        String unpublishedPid = "doi:10.666/FAKE/unpublished1";
+        unpublished.setGlobalId(new GlobalId(unpublishedPid));
+        Timestamp nullPublicationDateToIndicateNotPublished = null;
+        unpublished.setPublicationDate(nullPublicationDateToIndicateNotPublished);
+        datasets.add(unpublished);
+
+        Dataset harvested = new Dataset();
+        String harvestedPid = "doi:10.666/FAKE/harvested1";
+        harvested.setGlobalId(new GlobalId(harvestedPid));
+        harvested.setPublicationDate(new Timestamp(new Date().getTime()));
+        harvested.setHarvestedFrom(new HarvestingClient());
+        datasets.add(harvested);
+
+        Dataset deaccessioned = new Dataset();
+        String deaccessionedPid = "doi:10.666/FAKE/harvested1";
+        deaccessioned.setGlobalId(new GlobalId(deaccessionedPid));
+        deaccessioned.setPublicationDate(new Timestamp(new Date().getTime()));
+        List<DatasetVersion> datasetVersions = new ArrayList<>();
+        DatasetVersion datasetVersion = new DatasetVersion();
+        datasetVersion.setVersionState(DatasetVersion.VersionState.DEACCESSIONED);
+        datasetVersions.add(datasetVersion);
+        deaccessioned.setVersions(datasetVersions);
+        datasets.add(deaccessioned);
+
+        SiteMapUtil.updateSiteMap(dataverses, datasets);
+
+        Exception wellFormedXmlException = null;
+        try {
+            assertTrue(XmlValidator.validateXmlWellFormed("/tmp/sitemap.xml"));
+        } catch (Exception ex) {
+            System.out.println("Exception caught checking that XML is well formed: " + ex);
+            wellFormedXmlException = ex;
+        }
+        assertNull(wellFormedXmlException);
+
+        Exception notValidAgainstSchemaException = null;
+        try {
+            assertTrue(XmlValidator.validateXmlSchema("/tmp/sitemap.xml", new URL("https://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd")));
```

Comment:
Ahh I see we validate here. Do we know if-when google updates there schema whether they will support old versions? I could see value in general of validating against the schema on each run if it lets us know more quickly that google won't stop indexing us.